Makefile

```diff
@@ -15,6 +15,18 @@ endif
 
 check-variables:
 
+init:
+	test -n "$(SOURCE_AMI)"  # $$SOURCE_AMI
+	test -n "$(PROJECT)"  # $$PROJECT
+	test -n "$(ENVIRONMENT)"  # $$ENVIRONMENT
+	test -n "$(FUNCTION)"  # $$FUNCTION
+	test -n "$(PUPPET_REPO)"  # $$PUPPET_REPO
+	test -n "$(PACKER_PROFILE)" #$$PACKER_PROFILE
+
+	if [[ -d "puppet/" ]] && ! git --git-dir=./puppet/.git remote get-url origin | grep --quiet $(PUPPET_REPO); then rm -rf ./puppet; fi
+	if [[ ! -d "puppet/" ]]; then git clone -b $(GIT_BRANCH) $(PUPPET_REPO) puppet; fi
+	cd puppet && git pull && git checkout $(GIT_BRANCH) && git submodule update --init --recursive
+	if [[ -e "puppet/Puppetfile" ]]; then cd puppet; r10k puppetfile install; fi
 
 build:
```

Comment:
you can run the `init` task as a pre-requisite of `build`, so you don't repeat yourself.

```
build: init
...
```

this should do the trick